# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceChild.cpp
# Commit: 675785853d7a
# Full Hash: 675785853d7a40123826e1b47c49d68f87bd4bd0
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2024-12-17 17:00:49
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 5: Exempt third-party cookie blocking if the channel is on the exception list. r=cookie-reviewers,bvandersloot,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D223384
# ==============================================================================

diff -r de6a26d3e3df -r 675785853d7a netwerk/cookie/CookieServiceChild.cpp
--- a/netwerk/cookie/CookieServiceChild.cpp	Tue Dec 17 08:57:31 2024 +0000
+++ b/netwerk/cookie/CookieServiceChild.cpp	Tue Dec 17 08:57:31 2024 +0000
@@ -484,7 +484,8 @@
     moreCookies =
         parser.Parse(baseDomain, requireHostMatch, cookieStatus, cookieString,
                      dateHeader, true, isForeignAndNotAddon, mustBePartitioned,
-                     storagePrincipalOriginAttributes.IsPrivateBrowsing());
+                     storagePrincipalOriginAttributes.IsPrivateBrowsing(),
+                     loadInfo->GetIsOn3PCBExceptionList());
     if (!parser.ContainsCookie()) {
       continue;
     }
